src/Riverline/DynamoDB/Connection.php

```diff
@@ -59,15 +62,25 @@ public function __construct($key, $secret, $region, $version = 'latest')
      * @return Connection
      * @throws ConfigurationException
      */
-    static public function CreateFromDynamoDbClient(DynamoDbClient $dynamoDbClient)
+    public static function CreateFromDynamoDbClient(DynamoDbClient $dynamoDbClient)
     {
-        $apiVersion = $dynamoDbClient->getApiVersion();
-        if($apiVersion != '2011-12-05') {
-            throw new ConfigurationException("Api version '{$apiVersion}' is not supported");
+        $apiVersion = $dynamoDbClient->getApi()->getApiVersion();
+
+        if ($apiVersion !== self::SUPPORTED_API_VERSION) {
+            throw new ConfigurationException("Api version '{$apiVersion}' is not supported, use " . self::SUPPORTED_API_VERSION);
         }
 
-        $connection = new Connection(null, null, 'us-east-1');
+        /** @var Credentials $credentials */
+        $credentials = $dynamoDbClient->getCredentials()->wait();
+
+        $connection = new Connection(
+            $credentials->getAccessKeyId(),
```

Comment:
Isn't this redundant? The first thing that happens on line 82 is you overwrite all of this by providing the client. That's why they passed `null` originally I believe.

If you wanted, it might make sense to just delete this method and then change the __construct function to take a client. That probably makes more sense in our platform (and make the corresponding change in DynamoService).

Then the comment above about API versions is redundant too.